apps/postgresql-server/bin/apply_migrations.sh

```diff
@@ -55,67 +29,8 @@ echo "New schema version: ${NEW_SCHEMA_VERSION}"
     -w \
     start
 
-# Read old (database) schema version
-OLD_SCHEMA_VERSION=$(psql -v ON_ERROR_STOP=1 -p "${TEMP_PORT}" -t -c \
-    "SELECT value FROM database_variables WHERE name = 'database-schema-version';" | \
-    xargs)
-
-validate_schema_version "${OLD_SCHEMA_VERSION}"
-
-echo "Old schema version: ${OLD_SCHEMA_VERSION}"
-
-# Run migration if needed
-if (( "${OLD_SCHEMA_VERSION}" == "${NEW_SCHEMA_VERSION}" )); then
-    echo "Schema is up-to-date, nothing to do."
-elif (( "${OLD_SCHEMA_VERSION}" > "${NEW_SCHEMA_VERSION}" )); then
-    echo "Schema is newer in the database, go write a migration and rebuild this image."
-    echo "old version: $OLD_SCHEMA_VERSION, new_version: $NEW_SCHEMA_VERSION"
-    exit 1
-else
-    echo "Upgrading from ${OLD_SCHEMA_VERSION} to ${NEW_SCHEMA_VERSION}..."
-    CONCAT_MIGRATION=$(mktemp /var/tmp/migration-${OLD_SCHEMA_VERSION}-${NEW_SCHEMA_VERSION}.XXXXXX)
-
-    # Run concatenated migration files in a single transaction
-    echo "BEGIN;" >> "${CONCAT_MIGRATION}"
-
-    for SCHEMA_VERSION in $(seq "${OLD_SCHEMA_VERSION}" "$(( ${NEW_SCHEMA_VERSION} - 1 ))"); do
-        MIGRATION_FILENAME="mediawords-${SCHEMA_VERSION}-$(( ${SCHEMA_VERSION} + 1 )).sql"
-        MIGRATION_PATH="${MIGRATIONS_DIR}/${MIGRATION_FILENAME}"
-
-        if [ ! -f "${MIGRATION_PATH}" ]; then
-            echo "Migration at ${MIGRATION_PATH} does not exist."
-            exit 1
-        fi
-        
-        cat "${MIGRATION_PATH}" >> "${CONCAT_MIGRATION}"
-
-    done
-
-    # Add shim which verifies that schema version has been set to correct value
-    cat >> "${CONCAT_MIGRATION}" << EOF
-
-        DO \$\$
-        BEGIN
-            IF NOT EXISTS (
-                SELECT 1
-                FROM database_variables
-                WHERE name = 'database-schema-version'
-                  AND value = '${NEW_SCHEMA_VERSION}'
-            ) THEN
-                RAISE EXCEPTION 'Schema version has not been updated after applying the migration';
-            END IF;
-        END\$\$;
-
-EOF
-
-    # Commit transaction
-    echo "COMMIT;" >> "${CONCAT_MIGRATION}"
-
-    # Apply migration
-    psql -v ON_ERROR_STOP=1 -p "${TEMP_PORT}" -f "${CONCAT_MIGRATION}"
-
-    rm "${CONCAT_MIGRATION}"
-fi
+# apply migrations
```

Comment:
It doesn't seem to work on second run of the container when the migrations *do* have to be applied with the server running on a temporary port before restarting it on the public (5432) port:

```bash
➜ docker run -it gcr.io/mcback/postgresql-server:jot-pgmigrate bash
postgres@4250ea85468a:/$ /opt/postgresql-server/bin/postgresql.sh
Skipping schema migrations on first run...
2021-06-08 08:15:58 EDT [9-1] LOG:  starting PostgreSQL 13.3 (Ubuntu 13.3-1.pgdg20.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, 64-bit
2021-06-08 08:15:58 EDT [9-2] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2021-06-08 08:15:58 EDT [9-3] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2021-06-08 08:15:58 EDT [27-1] LOG:  database system was shut down at 2021-06-08 08:15:31 EDT
2021-06-08 08:15:58 EDT [9-4] LOG:  database system is ready to accept connections
^C2021-06-08 08:16:02 EDT [9-5] LOG:  received fast shutdown request
2021-06-08 08:16:02 EDT [9-6] LOG:  aborting any active transactions
2021-06-08 08:16:02 EDT [9-7] LOG:  background worker "logical replication launcher" (PID 33) exited with exit code 1
2021-06-08 08:16:02 EDT [28-1] LOG:  shutting down
2021-06-08 08:16:03 EDT [9-8] LOG:  database system is shut down

postgres@4250ea85468a:/$ /opt/postgresql-server/bin/postgresql.sh
Applying schema migrations...
waiting for server to start....2021-06-08 08:16:04 EDT [47-1] LOG:  starting PostgreSQL 13.3 (Ubuntu 13.3-1.pgdg20.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, 64-bit
2021-06-08 08:16:04 EDT [47-2] LOG:  listening on IPv4 address "0.0.0.0", port 12345
2021-06-08 08:16:04 EDT [47-3] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.12345"
2021-06-08 08:16:04 EDT [48-1] LOG:  database system was shut down at 2021-06-08 08:16:02 EDT
2021-06-08 08:16:04 EDT [47-4] LOG:  database system is ready to accept connections
 done
server started
Traceback (most recent call last):
  File "/usr/local/bin/pgmigrate", line 8, in <module>
    sys.exit(_main())
  File "/usr/local/lib/python3.8/dist-packages/pgmigrate.py", line 889, in _main
    config = get_config(args.base_dir, args)
  File "/usr/local/lib/python3.8/dist-packages/pgmigrate.py", line 816, in get_config
    conf = conf._replace(conn_instance=_create_connection(conf))
  File "/usr/local/lib/python3.8/dist-packages/pgmigrate.py", line 171, in _create_connection
    conn = _create_raw_connection(
  File "/usr/local/lib/python3.8/dist-packages/pgmigrate.py", line 163, in _create_raw_connection
    conn = psycopg2.connect(conn_string, connection_factory=LoggingConnection)
  File "/usr/local/lib/python3.8/dist-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?
could not connect to server: Cannot assign requested address
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 5432?

postgres@4250ea85468a:/$ ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
postgres       1  0.0  0.0   7420  3776 pts/0    Ss   08:15   0:00 bash
postgres      47  0.0  1.7 5836140 277276 ?      Ss   08:16   0:00 /usr/lib/postgresql/13/bin/postgres -D /var/lib/postgresql/13/main -c config_file=/etc/po
postgres      49  0.0  0.0 5836140 4196 ?        Ss   08:16   0:00 postgres: 13/main: checkpointer 
postgres      50  0.0  0.3 5836304 55752 ?       Ss   08:16   0:00 postgres: 13/main: background writer 
postgres      51  0.0  0.1 5836140 22112 ?       Ss   08:16   0:00 postgres: 13/main: walwriter 
postgres      52  0.0  0.0 5837116 8616 ?        Ss   08:16   0:00 postgres: 13/main: autovacuum launcher 
postgres      53  0.0  0.0  70584  5516 ?        Ss   08:16   0:00 postgres: 13/main: stats collector 
postgres      54  0.0  0.0 5836952 6496 ?        Ss   08:16   0:00 postgres: 13/main: logical replication launcher 
postgres      62  0.0  0.0   9208  3228 pts/0    R+   08:18   0:00 ps aux
```

I think the port is misconfigured here (maybe).

Reply:
Ah right you are, I had to pass a connection string argument with `port=12345` to override the default `5432`.